torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -86,11 +87,12 @@ def __init__(self, args: ModelArgs):
         self.dropout_p = args.dropout_p
         self.resid_dropout = nn.Dropout(args.dropout_p)
         self.use_attn_mask = args.use_attn_mask
+        self.dtype = args.dtype
 
-        self.wq = nn.Linear(args.dim, args.dim, bias=False)
-        self.wk = nn.Linear(args.dim, args.dim, bias=False)
-        self.wv = nn.Linear(args.dim, args.dim, bias=False)
-        self.wo = nn.Linear(args.dim, args.dim, bias=False)
+        self.wq = nn.Linear(args.dim, args.dim, bias=False, dtype=args.dtype)
```

Comment:
instead of passing dtype to every layer initialization, can we simply do:

`Transformer.to(dtype=args.dtype)`

instead?

Reply:
Yeah good point, sorry I didn't know this API. In this case, I think it is unnecessary to pass `dtype` into `ModelArgs`.